src/main/java/uk/gov/hmcts/darts/usermanagement/service/impl/SecurityGroupServiceImpl.java

```diff
@@ -82,6 +96,87 @@ public List<SecurityGroupWithIdAndRole> getSecurityGroups(List<Integer> roleIds,
             .map(securityGroupCourthouseMapper::mapToSecurityGroupWithIdAndRoleWithCourthouse).toList();
     }
 
+    @Transactional
+    @Override
+    public SecurityGroupWithIdAndRoleAndUsers modifySecurityGroup(Integer securityGroupId, SecurityGroupPatch securityGroupPatch) {
+
+        Optional<SecurityGroupEntity> securityGroupEntityOptional = securityGroupRepository.findById(securityGroupId);
+
+        if (securityGroupEntityOptional.isPresent()) {
+            SecurityGroupEntity securityGroupEntity = securityGroupEntityOptional.get();
+            updateSecurityGroupEntity(securityGroupPatch, securityGroupEntity);
+            securityGroupRepository.saveAndFlush(securityGroupEntity);
+            return securityGroupCourthouseMapper.mapToSecurityGroupWithCourthousesAndUsers(securityGroupEntity);
+        } else {
+            //throw a 404 not found
+            throw new DartsApiException(
+                UserManagementError.SECURITY_GROUP_NOT_FOUND,
+                String.format("Security group id %d not found", securityGroupId));
+        }
+
+    }
+
+    protected void updateSecurityGroupEntity(SecurityGroupPatch securityGroupPatch, SecurityGroupEntity securityGroupEntity) {
+        validate(securityGroupPatch, securityGroupEntity);
+
+        String name = securityGroupPatch.getName();
+        String displayName = securityGroupPatch.getDisplayName();
+
+        if (name != null) {
```

Comment:
Wondering for these updates `name / displayName / description` - does it make sense to allow blank here?
I guess an admin user could force the update each time just adding blank spaces to name if they wanted to save the record.
I think we should use `StringUtils.isNotBlank` - what do you think @chrisbellingham-hmcts ?

Reply:
Agree on non blank for name and displayName (although this should already be enforced at the openapi layer, but belt and braces?).

Description is optional, so blank is an allowable value as sending a blank string here this is the only way to "clear" an existing description.

Reply:
To that end @chrisdavecm , can you please relax the constraint for the `SecurityGroupDescription` schema type in `usermanagement.yaml` to have ` minLength: 0`, rather than `1`?